source/components/molecules/FilePicker/FilePicker.tsx

```diff
@@ -46,10 +46,14 @@ const FilePicker: React.FC<Props> = ({
     file: File,
     baseName: string,
     suffix: string
-  ): File => ({
-    ...file,
-    externalDisplayName: `${baseName}_${suffix}`,
-  });
+  ): File => {
+    const [, fileExtension] = file.mime.split("/");
```

Comment:
Consider using a lut instead:

```typescript
const mimeExtensionLookup = {
    "application/pdf": ".pdf",
    "image/jpeg": ".jpg",
    "image/png": ".png",
}

const extension = mimeExtensionLookup[file.mime] ?? "";

return {
    ...file,
    externalDisplayName: `${baseName}_${suffix}${extension}`,
};
```

this would also handle unknown mime types as extensionless filenames correctly.

Reply:
Will add preferredFileName directly in image/pdf upload functions to avoid modifying an already existing object.